File: check_tax.py
from web3 import Web3, HTTPProvider
from json import loads

w3 = Web3(HTTPProvider('https://bsc-dataseed.binance.org/'))

# 读取 contract_data.js 里的 ABI
with open(r'C:\Users\Administrator\WorkBuddy\2026-06-17-21-17-01\contract_data.js', 'r', encoding='utf-8') as f:
    src = f.read()
# 提取 ABI
abi_start = src.find('"ABI":') + 6
abi_end = src.find('],', abi_start) + 1
ABI = loads(src[abi_start:abi_end+1])

contract_address = '0x6Fe41178284F5651D09d56CC6A6D3D87C60AC555'
c = w3.eth.contract(address=contract_address, abi=ABI)

for fn in ['buyTaxBps', 'sellTaxBps', 'MAX_TAX']:
    try:
        v = c.functions[fn]().call()
        print(f'{fn} = {v}  (={v/100}%)' if 'Tax' in fn or 'MAX' in fn else f'{fn} = {v}')
    except Exception as e:
        print(f'{fn}: ERROR - {e}')

File: check_tax.js
const { ethers } = require('ethers');
const fs = require('fs');
const src = fs.readFileSync('./contract_data.js', 'utf8');
eval(src.replace('const CONTRACT_DATA', 'var CONTRACT_DATA'));
const ABI = CONTRACT_DATA.ABI;

const provider = new ethers.JsonRpcProvider('https://bsc-dataseed.binance.org/');
const c = new ethers.Contract('0x6Fe41178284F5651D09d56CC6A6D3D87C60AC555', ABI, provider);

async function main() {
  const [buyTax, sellTax, maxTax] = await Promise.all([
    c.buyTaxBps(), c.sellTaxBps(), c.MAX_TAX()
  ]);
  console.log('链上实际税率：');
  console.log('  buyTaxBps  =', buyTax.toString(), '→', (buyTax / 10000 * 100).toFixed(2) + '%');
  console.log('  sellTaxBps =', sellTax.toString(), '→', (sellTax / 10000 * 100).toFixed(2) + '%');
  console.log('  MAX_TAX    =', maxTax.toString());
}
main().catch(console.error);
